var/spack/repos/builtin/packages/py-postcactus/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyPostcactus(PythonPackage):
+    """This repository contains a Python package named PostCactus for
+    postprocessing data from numerical simulations performed with the Einstein
+    Toolkit."""
+
+    homepage = "https://github.com/wokast/PyCactus"
+    url = "https://github.com/wokast/PyCactus/archive/refs/tags/2.2.zip"
+
+    version("2.2", sha256="303108835d7652b37f43871c735b99dc7a60c28a24de35a09e9d2bb0f28f93fb")
+
+    # pyproject.toml
+    depends_on("py-setuptools@40.6.0:", type="build")
+    # setup.cfg
+    depends_on("py-h5py", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"))
+    depends_on("py-future", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
+    depends_on("py-tables", type=("build", "run"))
+    # recommended README.md
+    depends_on("py-jupyter", type=("build", "run"))
+    depends_on("py-sphinx", type=("build", "run"))
+    depends_on("vtk", type=("build", "run"))
```

Comment:
I would personally not include these recommended dependencies. Jupyter is not used by the library, although some people might like to use it in general. Sphinx is only needed to build the docs, which we aren't doing. Vtk is actually used by the library, but it's a heavy optional dependency that a lot of people may run into trouble building.

Reply:
I agree with adamjstewart with the caveat that 29 and 31 should be an optional variant.

Reply:
@sidpbury @alex391 What about removing the depends on sphinx which I think is acknowledged and making the other two optional variants?

I hope you don't mind me setting this to draft to prevent GitLab CI resources?

You can set it to "ready for review" again once this is answered or handled of course!